src/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -204,7 +206,15 @@ private boolean processAttributeElement(HttpMessage message, int depth, String b
 			return false;
 		}
 
-		processURL(message, depth, localURL, baseURL);
+		if (!attributeName.equalsIgnoreCase("ping")) {
+			processURL(message, depth, localURL, baseURL);
+		} else {
+			for (String pingURL : localURL.split(" ")) {
+				if (!pingURL.equals(" ")) {
```

Comment:
Since the `ping` attribute allows multiple URLs separated by space I need an elegant way to handle the `(same page)` scenario while also handling the fact that the space delimited list might have multiple spaces which are irrelevant. Any ideas?

Also need to ensure that the ping URLs start with http(s).

Note, I haven't yet added the UnitTests for area handling until I get this part sorted.

Reply:
I've been thinking about the `(same page)` scenario(s) vs. handling multiple spaces in the list. I think the best that can be done is:
* Check the length of the array resulting from split. If it's a single item then process as normal (`(same page)` scenario will work). If it's multiple then only process if not space (as above) and a `(same page)` scenario might be missed.
  * It seems unlikely that many people/frameworks would choose to process the pings (by snooping headers/POST content etc at the same endpoint as is hosting the content), and do so by choosing to not specify the destination at all (ex: `ping=""`). [Not impossible but perhaps improbable?]

If we just document this behavior, is that sufficient? Or implement it and see what kind of feedback we get?

Reply:
What do you mean by `(same page)`?

Reply:
If you look at the unittests, there's a `(same page)` variant that's essentially an empty string `""`

Reply:
Sorry, missed that. In that case the list of strings is empty (if I'm reading https://infra.spec.whatwg.org/#split-on-ascii-whitespace correctly), so no pings to handle.

Reply:
It's not necessary to ignore non-HTTP(S) URLs, the spider already does that.